common/src/print.cpp

```diff
@@ -110,6 +109,13 @@ useColoredOutput (FILE *stream)
   return colored.get ();
 }
 
+pcl::console::Logger&
+pcl::console::Logger::getInstance()
+{
+  static Logger instance;
+  return instance;
+}
+
 ////////////////////////////////////////////////////////////////////////////////
```

Comment:
```c++
template <typename Functor>
void pcl::console::Logger::setCallback(Functor&& callback){
      getInstance().setCallback(std::move(callback));
}
```
DISCLAIMER: draft, untested code.
@larshg wouldn't this work? Would the template be a problem there?

Reply:
You want a free function like:
```

    template <typename Functor>
    void
    setCallback(Functor&& callback)
    {
      Logger::getInstance().setCallback(std::move(callback));
    }
```

so you can call:
`pcl::console::setCallback(...);`

Yes, thats possible.